File: 프로그래머스/0/181916. 주사위 게임 3/주사위 게임 3.java
class Solution {
    public int solution(int a, int b, int c, int d) {
        if(a==b && b==c && c==d) return 1111*a;
        if(a==b && b==c && c!=d) return (int)Math.pow(10*c+d,2);
        if(a==b && b==d && d!=c) return (int)Math.pow(10*d+c,2);
        if(a==c && c==d && d!=b) return (int)Math.pow(10*d+b,2);
        if(b==c && c==d && d!=a) return (int)Math.pow(10*b+a,2);
        if(a==b && c==d && a!=c) return (a+c)*Math.abs(a-c);
        if(a==c && b==d && a!=b) return (a+b)*Math.abs(a-b);
        if(a==d && b==c && a!=b) return (a+b)*Math.abs(a-b);
        if(a==b && c!=d) return c*d;
        if(a==c && b!=d) return b*d;
        if(a==d && b!=c) return b*c;
        if(b==c && a!=d) return a*d;
        if(b==d && a!=c) return a*c;
        if(c==d && a!=b) return a*b;
        return Math.min(a,Math.min(b,Math.min(c,d)));
    }
}

File: 프로그래머스/0/181916. 주사위 게임 3/주사위 게임 3.py
from collections import Counter

def solution(a, b, c, d):
    cnt = list(Counter([a,b,c,d]).items())
    
    if len(cnt) == 1:
        return cnt[0][0] * 1111
    elif len(cnt) == 2:
        if cnt[0][1] == 3 or cnt[1][1] == 3:
            return (10*cnt[0][0]+cnt[1][0])**2 if(cnt[0][1]==3) else (10*cnt[1][0]+cnt[0][0])**2
        else:
            return (cnt[0][0]+cnt[1][0])*(abs(cnt[0][0]-cnt[1][0]))
    elif len(cnt) == 3:
        result = 1
        for i in range(3):
            if cnt[i][1] == 1:
                result *= cnt[i][0]
        return result
    else:
        return min(cnt[0][0], cnt[1][0], cnt[2][0], cnt[3][0])